install/operator/pkg/generator/assets/install/grant_public_api_cluster_role.yml.tmpl

```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: syndesis-{{.Namespace}}-auth-delegator
+  labels:
+    app: syndesis
+    syndesis.io/app: syndesis
+    syndesis.io/type: infrastructure
+    syndesis.io/component: syndesis-public-oauthproxy
+subjects:
+- kind: ServiceAccount
+  name: syndesis-public-oauthproxy
+  namespace: {{.Namespace}}
+roleRef:
+  kind: ClusterRole
+  name: system:auth-delegator
+  apiGroup: rbac.authorization.k8s.io
```

Comment:
I have to point out that this is generally a bad practice towards OLM integration because, they are picky with the verbs they allow and it can lead to the operator not being allowed to bind the `ClusterRole` in OLM if the `ClusterRole` has any verb other than `assign get list watch create update patch delete deletecollection initialize use`. I rather have a new `Role` with all the specific permissions you need, and a `RoleBinding` for it, which in my opinion also adds clarity of what is being granted.

**BUT**, since this ClusterRole is so slim, I think it is fine to leave it this way.

Nevertheless you have to add those permissions to https://github.com/syndesisio/syndesis/blob/684df965f085499789b8033385abb63027e4b255/install/operator/pkg/generator/assets/install/role.yml.tmpl#L10

Reply:
I have added a ClusterRole to avoid referencing the OpenShift system:auth-delgator role if it's going to be an issue with OLM.